manager/integration/tests/test_basic.py

```diff
@@ -3701,3 +3702,66 @@ def test_expand_pvc_with_size_round_up(client, core_api, volume_name):  # NOQA
 
     client.delete(volume)
     wait_for_volume_delete(client, volume_name)
+
+
+def test_backuptarget_available_during_engine_image_not_ready(client, apps_api):  # NOQA
+    """
+    Test backup target available during engine image not ready
+
+    1. Set backup target URL to S3 and NFS respectively
+    2. Set poll interval to 0 and 300 respectively
+    3. Restart engine image
+    4. Configures backup target during engine image Pods restart
+    5. Check backup target status.available=true
+    6. After engine image Pods are Ready, unset backup target configuration
+    7. Check backup target status.available=false
+    """
+    backupstores = common.get_backupstore_url()
+    for backupstore in backupstores:
+        url = ""
+        cred_secret = ""
+        if common.is_backupTarget_s3(backupstore):
+            backupsettings = backupstore.split("$")
+            url = backupsettings[0]
+            cred_secret = backupsettings[1]
+        elif common.is_backupTarget_nfs(backupstore):
+            url = backupstore
+            cred_secret = ""
+        else:
+            raise NotImplementedError
+
+        poll_intervals = ["0", "300"]
+        for poll_interval in poll_intervals:
+            set_backupstore_poll_interval(client, poll_interval)
+
+            # Restart engine image DaemonSet
+            default_img = common.get_default_engine_image(client)
+            ds_name = "engine-image-" + default_img.name
+            apps_api.delete_namespaced_daemon_set(
+                ds_name, common.LONGHORN_NAMESPACE)
+            time.sleep(1)  # Wait for the engine image Pods be deleted
+
+            # Set a valid backup target during the engine image Pods restart
+            backupsettings = backupstore.split("$")
+            set_backupstore_url(client, url)
+            set_backupstore_credential_secret(client, cred_secret)
+            time.sleep(2)  # Wait for the backup_target_controller
+            bt = common.find_backup_target_default(client)
+            if not bt.available:
+                raise Exception(
+                    'BackupTarget status.available should be True')
```

Comment:
Use a retry loop to validate the backup target becoming available. Similar below